src/libcrun/cgroup-utils.c

```diff
@@ -71,6 +71,41 @@ libcrun_cgroups_create_symlinks (int dirfd, libcrun_error_t *err)
   return 0;
 }
 
+int
+make_cgroup_threaded (const char *path, libcrun_error_t *err)
+{
+  cleanup_free char *cgroup_path_type = NULL;
+  int ret;
+
+  path = consume_slashes (path);
+
+  if (path == NULL || path[0] == '\0')
+    return 0;
+
+  ret = append_paths (&cgroup_path_type, err, CGROUP_ROOT, path, "cgroup.type", NULL);
+  if (UNLIKELY (ret < 0))
+    return ret;
+
+  ret = write_file (cgroup_path_type, "threaded", strlen ("threaded"), err);
+  if (UNLIKELY (ret < 0 && errno == EOPNOTSUPP))
+    {
+      cleanup_free char *buffer = xstrdup (path);
+      const char *parent = consume_slashes (dirname (buffer));
+      if (parent[0])
+        {
+
+          crun_error_release (err);
+
+          ret = make_cgroup_threaded (parent, err);
+          if (ret < 0)
+            return ret;
+
+          return write_file (cgroup_path_type, "threaded", strlen ("threaded"), err);
```

Comment:
Not sure but I think this should not be needed we are already marking domain cgroup at `cgroup_path_type` as `threaded` in the above line.

Reply:
you are right.  I'll drop it

Reply:
no sorry, I think it is correct :)  We are retrying again after making the parent cgroup threaded